service/history/task/cross_cluster_task.go

```diff
@@ -470,6 +486,88 @@ func (t *crossClusterSourceTask) GetCrossClusterRequest() (request *types.CrossC
 	return request, nil
 }
 
+func (t *crossClusterSourceTask) VerifyTaskFailoverVersion(
+	mutableState execution.MutableState,
+	taskInfo *persistence.CrossClusterTaskInfo,
+) error {
+	lastWriteVersion, err := mutableState.GetLastWriteVersion()
+	if err != nil {
+		return err
+	}
+	ok, err := verifyTaskVersion(t.shard, t.logger, taskInfo.DomainID, lastWriteVersion, taskInfo.Version, taskInfo)
+	if err != nil || !ok {
+		return err
+	}
+	return nil
```

Comment:
if `ok == false` and `err == nil`, the function will return `nil`. Then the execution for the caller will continue, while we should skip the task by returning `nil, processingState, nil` in caller to indicate the task is a no-op I think.